examples/run_examples.sh

```diff
@@ -18,49 +44,200 @@ fi
 mpi_exec="${MIRGE_MPI_EXEC}"
 mpi_launcher="${MIRGE_PARALLEL_SPAWNER}"
 
-cd $examples_dir
-
 export OMPI_ALLOW_RUN_AS_ROOT=1
 export OMPI_ALLOW_RUN_AS_ROOT_CONFIRM=1
 
-for example in *.py
+for example in "${example_list[@]}"
 do
-    if [[ $example == "wave-nompi.py" ]]; then
-        echo "Skipping $example"
-        continue
+    example_name="${example%.py}"
+
+    # FIXME: Let's still test the serial examples
+    if [[ $example_name == *"nompi"* ]]; then
+        echo "*** Skipping serial example (for now): $example_name"
+        continue        
     fi
 
-    TOL_LAZY=1e-04
-    TOL_NUMPY=1e-05
 
-    if [[ $example == "thermally-coupled.py" ]]; then
-        echo "Setting tolerance=1 for $example"
-        TOL_LAZY=1
-        TOL_NUMPY=1
-    fi
+    # FIXME: The tolerances are really high
+
+    # Should be this:
+    #  TOL_LAZY=1e-12
+    #  TOL_NUMPY=1e-12
+
+    # But this is required to make them all pass
+    TOL_LAZY=1e-9
+    TOL_NUMPY=1e-9
 
+    # FIXME: Have to ignore "rhs", and "gradients" to make
+    # this example pass.
+    # if [[ $example == "thermally-coupled.py" ]]; then
+    #    echo "Setting tolerance=1 for $example"
+    #    TOL_LAZY=1
+    #    TOL_NUMPY=1
+    # fi
 
     date
     printf "***\n***\n"
 
-    ${mpi_exec} -n 2 $mpi_launcher python -m mpi4py $example --casename ${example}-eager
-    ${mpi_exec} -n 2 $mpi_launcher python -m mpi4py $example --casename ${example}-lazy --lazy
-    ${mpi_exec} -n 2 $mpi_launcher python -m mpi4py $example --casename ${example}-numpy --numpy
+    declare -A test_results
```

Comment:
Does this work on MacOS? Associative arrays are a bash 4 feature I think, while macOS has version 3

Reply:
Not out of the box, but it works in CI@github.  It also works on Lassen.  I brew installed bash and you get bash 5, then the script works on my Mac when I change the shbang to /opt/homebrew/bin/bash.

Reply:
removed this requirement in [1cc955a](https://github.com/illinois-ceesd/mirgecom/pull/951/commits/1cc955ad740f6b1a2540da223ddd31bc92428cfa)